packages/auth-express/readme.md

```diff
@@ -0,0 +1,397 @@
+
+```ts
+const builtinRouter = Router()
+  .get("/signin", auth.builtin.signIn)
+  .get("/signup", auth.builtin.signUp)
+  .get(
+    "/callback",
+    auth.builtin.callback,
+    (req: expressAuth.CallbackRequest, res, next) => {
+      if (req.isSignUp) {
+        return res.redirect("/onboarding");
+      }
+
+      res.redirect("/");
+    }
+  );
+
+app.use("/auth", builtinRouter);
+```
+
+#### Custom UI: Email and password
+
+```ts
+const emailPasswordRouter = Router()
+  .post(
+    "/signin",
+    auth.emailPassword.signIn,
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/");
+    }
+  )
+  .post(
+    "/signup",
+    auth.emailPassword.signUp(
+      // URL of the verify endpoint configured below
+      "http://localhost:3000/auth/email-password/verify"
+    ),
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/onboarding");
+    }
+  )
+  .post(
+    "/verify",
+    auth.emailPassword.verify,
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/");
+    }
+  )
+  .post(
+    "/send-password-reset-email",
+    auth.emailPassword.sendPasswordResetEmail(
+      // URL of the reset password endpoint configured below
+      "http://localhost:3000/auth/email-password/reset-password"
+    ),
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/email-success");
+    }
+  )
+  .post(
+    "/reset-password",
+    auth.emailPassword.resetPassword,
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/email-success");
+    }
+  )
+  .post(
+    "/resend-verification-email",
+    auth.emailPassword.resendVerificationEmail,
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/");
+    }
+  );
+
+app.use("/auth/email-password", emailPasswordRouter);
+```
+
+#### Custom UI: OAuth
+
+```ts
+const oAuthRouter = Router()
+  .get(
+    "/",
+    auth.oAuth.redirect(
+      // URL of the callback endpoint configured below
+      "http://localhost:3000/auth/oauth/callback"
+    )
+  )
+  .get(
+    "/callback",
+    auth.oAuth.callback,
+    (req: expressAuth.AuthRequest, res) => {
+      res.redirect("/");
+    }
+  );
+
+app.use("/auth/oauth", oAuthRouter);
+```
+
+### Error handling
+
+If an error occurs during the authentication flow, it will be passed to the Express error handler. You can use Express error handlers to handle errors either on individual routes by adding the error handlers to the middleware arrays or in your route definitions, or define a router-wide error handler. Any Express error handling pattern is available here, but let's examine a quick example of handling error with the built-in UI flow:
+
+```ts
+const builtinRouter = auth.createBuiltinRouter({
+  callback: [
+    (req: expressAuth.CallbackRequest, res, next) => {
+      if (req.isSignUp) {
+        return res.redirect("/onboarding");
+      }
+
+      res.redirect("/");
+    },
+    (error: any, req, res, next) => {
+      res.redirect(`/error?${encodeURIComponent(error.message)}`);
```

Comment:
```suggestion
      res.redirect(`/signin?error=${encodeURIComponent(error.message)}`);
```

Reply:
This is the built-in UI so any errors that make it outside of the typical error handling in the built-in UI are probably fatal, so I intended this to redirect to back to the actual application. Good catch about missing the query parameter tho 🙈